frontend/src/test/java/starter/project/frontend/CarControllerTest.java

```diff
@@ -1,21 +1,56 @@
 package starter.project.frontend;
 
-import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.test.web.client.TestRestTemplate;
-import org.springframework.boot.web.server.LocalServerPort;
-import org.springframework.http.ResponseEntity;
+import starter.project.domain.Car;
+import starter.project.domain.Doors;
+import starter.project.domain.Engine;
+import starter.project.domain.Wheels;
+import java.util.ArrayList;
+import java.util.List;
+import static org.junit.Assert.assertEquals;
 
-import java.net.URL;
-import static org.junit.jupiter.api.Assertions.*;
-import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
 
-@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 public class CarControllerTest {
-    @LocalServerPort
-    private int port;}
-//    @Autowired
-//    //private TestRestTemplate restTemplate
+
+    CarController controller = new CarController();
+
+    @Test
+    public void newCarsAAreAddedToMap() {
+
+        Car car = Car.builder().id(15).build();
+        controller.addCar(car);
+        assertEquals(1, controller.map.size());
```

Comment:
I think that the Map in this case should be "injected" via the constructor and made a private variable. This limits the scope and access to the map, but also makes it available for testing. 

This is called Dependency Injection.

https://stackoverflow.com/a/130862